src/client/headers/shared/containers.hpp

```diff
@@ -232,15 +261,31 @@ namespace intercept::types {
             }
             return rcount;
         }
+
+        /**
+         * @brief deletes object
+         * 
+         */
         void destruct() const noexcept {
             delete const_cast<refcount*>(this);
         }
+        /**
+         * @brief wrapper for destruct
+         * 
```

Comment:
```suggestion
         * @brief wrapper for destruct
```
"wrapper" is not really correct.
This is a customization point so sub classes can run custom code before destruction, which is why its virtual

Reply:
Ah that's something I must sleep through in my oop class